### ISSUE TITLE
Count is not the same as findAll

### ISSUE THREAD

**Issue by tomfun** [sequelize/sequelize#7852]
## What you are doing?
_Post a minimal code sample that reproduces the issue, including models and associations_

```js
        // Event -- m:1 --> Application
        // Application -- n:m -> User
        const Event = this.__model;
        const appAssoc = Event.associations.application;
        const Application = appAssoc.target;

        const userAssoc = Application.associations.users;
        const User = userAssoc.target;
        // I want to get all events, which belongs to applications, with user with id=1
        // I am a user with id=1, I want to see events for my applications
        const sequelizeOpts = {
            offset: 0,
            limit: 2,
            order: null,
            where: {},
            include: [
                {
                    association: appAssoc,
                    model: Application,
                    as: 'application',
                    // ! required: true ! commented
                    include: [
                        {
                            association: userAssoc,
                            model: User,
                            as: 'users',
                            where: {id: 1}
                        }
                    ]
                }
            ],
            attributes: {include: [], exclude: []}
        };
        const events = yield Event.findAndCountAll(sequelizeOpts);
        console.log('\r\nevents.rows.length\r\n', events.rows.length)
        console.log('\r\nevents.count\r\n', events.count)
```
There are only 3 items (Event) and no one is mine (I'm not the owner of them)

## What do you expect to happen?
_I want that `count` to will be the same as real rows length, returned by `findAll`!_

## What is actually happening?
_But the output was:!_
```txt
events.rows.length
  0
events.count
 3
```

_Output SQL:_

Count:
```sql
SELECT count(`Event`.`id`) AS `count`
FROM `events` AS `Event`
  -- LEFT, not INNER
  LEFT OUTER JOIN (`applications` AS `application` INNER JOIN
  (`application_owners` AS `application->users->ApplicationOwner` INNER JOIN `users` AS `application->users`
      ON `application->users`.`id` = `application->users->ApplicationOwner`.`user_id`)
    ON `application`.`id` = `application->users->ApplicationOwner`.`app_id` AND `application->users`.`id` = 1)
    ON `Event`.`app_id` = `application`.`id`;
```

FindAll:
```sql
SELECT
  `Event`.*,
  `application->users`.`id`                        AS `application.users.id`,
  `application->users`.`username`                  AS `application.users.username`,
  `application->users`.`type`                      AS `application.users.type`,
  `application->users`.`is_active`                 AS `application.users.is_active`,
  `application->users->ApplicationOwner`.`user_id` AS `application.users.ApplicationOwner.user_id`,
  `application->users->ApplicationOwner`.`app_id`  AS `application.users.ApplicationOwner.app_id`
FROM (SELECT
        `Event`.`id`,
        -- I stripped other fields --
        -- ... --
        `Event`.`created_at`,
        `Event`.`updated_at`,
        `application`.`id`                  AS `application.id`
        -- ... --
      FROM `events` AS `Event` LEFT OUTER JOIN `applications` AS `application` ON `Event`.`app_id` = `application`.`id`
      WHERE (SELECT `Application`.`id`
             FROM `applications` AS `Application` INNER JOIN
               (`application_owners` AS `users->ApplicationOwner` INNER JOIN `users` AS `users`
                   ON `users`.`id` = `users->ApplicationOwner`.`user_id`)
                 ON `Application`.`id` = `users->ApplicationOwner`.`app_id` AND `users`.`id` = 1
             WHERE `Event`.`app_id` = `Application`.`id`
             LIMIT 1) IS NOT NULL
      LIMIT 0, 2) AS `Event`
  -- @@@ INNER not the LEFT @@@ ---
  INNER JOIN
  (`application_owners` AS `application->users->ApplicationOwner` INNER JOIN `users` AS `application->users`
      ON `application->users`.`id` = `application->users->ApplicationOwner`.`user_id`)
    ON `application.id` = `application->users->ApplicationOwner`.`app_id` AND `application->users`.`id` = 1;
```

I am not sure what is the right:  
From one point of view - the `count` is ok, but `findAll` fails. It is because I don't use `required: true` and there is no `where` in the *include* part, so in the result SQL must be **left join** . So as I said, `count` is ok - there is *left join*.  
From the other point of view (what I must do for the moment) - `count` fails, but `findAll` is ok. It is because this is what was before - in the sequelize v3.

__Dialect:__ mysql
__Database version:__ 5.6.33-0ubuntu0.14.04.1
__Sequelize version:__ 4.2.0

**Comment by Alexsey** [comment]
Looks like related to https://github.com/sequelize/sequelize/issues/7344

**Comment by tomfun** [comment]
@Alexsey I am not sure about. In my situation generated *left join* instead of *inner join*.

**Comment by Alexsey** [comment]
@tomfun You are right, sorry, they are definitely not related

**Comment by L3V147H4N** [comment]
this is still an issue using version 4.8.3

**Comment by L3V147H4N** [comment]
adding `distinct: true` fixed it #2713
